editoast/editoast_models/src/model.rs

```diff
@@ -0,0 +1,97 @@
+use std::sync::LazyLock;
+
+use diesel::result::{DatabaseErrorInformation, DatabaseErrorKind};
+use regex::Regex;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("unique constraint violation \"{constraint}\" on column \"{column}\" with value \"{value}\"")]
+    UniqueViolation {
+        constraint: String,
+        column: String,
+        value: String,
+    },
+    #[error("check constraint violation of \"{constraint}\"")]
+    CheckViolation { constraint: String },
+    #[error(transparent)]
+    DatabaseError(#[from] crate::DatabaseError),
+}
+
+fn try_parse_unique_violation(
+    e: &Box<dyn DatabaseErrorInformation + Send + Sync>,
+) -> Option<Error> {
+    static RE: LazyLock<Regex> =
+        LazyLock::new(|| Regex::new(r#"duplicate key value violates unique constraint"#).unwrap());
+    if RE.is_match(e.message()) {
+        static RE: LazyLock<Regex> = LazyLock::new(|| {
+            Regex::new(r#"Key \(([^)]+)\)=\(([^)]+)\) already exists\."#).unwrap()
+        });
+        if let Some(captures) = RE.captures(e.details().expect("PostgreSQL should provide details"))
```

Comment:
No need for panic, especially since [this might actually return nothing](https://docs.rs/diesel/latest/diesel/result/trait.DatabaseErrorInformation.html#tymethod.details).

```suggestion
        if let Some(captures) = RE.captures(e.details().unwrap_or("Database did not provided details"))
```